statviz/src/views/Statistics/components/filter/TimeRangeSelect.tsx

```diff
@@ -0,0 +1,113 @@
+import { Box, Button, ButtonGroup, HStack, Heading } from "@chakra-ui/react";
+import DateField from "../../../../components/Form/DateField";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod/src/zod.js";
+import { useForm } from "react-hook-form";
+import { addDays, isBefore, subDays, subMonths } from "date-fns";
+import { useEffect } from "react";
+import { useSearchParams } from "react-router-dom";
+
+export const FilterCreatedOnFormScheme = z.object({
+  from: z
+    .date({
```

Comment:
maybe you want to check how zod works for custom validation (refine function) https://github.com/boxwise/boxtribute/blob/94da29211b2d7d5dd0db63930fb08eb448d1de15/react/src/views/BoxEdit/components/BoxEdit.tsx#L61